File: cinema-ticket-C#/Model/Screening.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace CinemaTicket.Model;

[Table("screenings")]
public class Screening
{
    public long Id { get; set; }

    [Required]
    public Cinema Cinema { get; set; } = null!;
    public long CinemaId { get; set; }

    [Required]
    public Movie Movie { get; set; } = null!;
    public long MovieId { get; set; }

    [Required]
    public DateTime StartDateTime { get; set; }

    [Column(TypeName = "decimal(10,2)")]
    [Range(typeof(decimal), "0.01", "9999999999.99")]
    public decimal TicketPrice { get; set; }

    public ICollection<Reservation> Reservations { get; set; } = new List<Reservation>();
    public ICollection<SeatHold> SeatHolds { get; set; } = new List<SeatHold>();

    [Timestamp]
    public byte[] RowVersion { get; set; } = Array.Empty<byte>();
}

File: Backend-Java/src/main/java/com/cinema/ticketsystem/model/Screening.java
package com.cinema.ticketsystem.model;

import jakarta.persistence.*;
import jakarta.validation.constraints.DecimalMin;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "screenings")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Screening {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "cinema_id", nullable = false)
    private Cinema cinema;
    
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "movie_id", nullable = false)
    private Movie movie;
    
    @NotNull
    @Column(nullable = false)
    private LocalDateTime startDateTime;
    
    @NotNull
    @DecimalMin("0.01")
    @Column(nullable = false, precision = 10, scale = 2)
    private BigDecimal ticketPrice;
    
    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @OneToMany(mappedBy = "screening", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Reservation> reservations = new ArrayList<>();
}
